plugins/redfish/fu-redfish-device.h

```diff
@@ -23,10 +23,29 @@ struct _FuRedfishDeviceClass {
 #define FU_REDFISH_DEVICE_FLAG_WILDCARD_TARGETS		"wildcard-targets"
 #define FU_REDFISH_DEVICE_FLAG_NO_MANAGER_RESET_REQUEST "no-manager-reset-request"
 
+typedef struct {
+	FwupdError error_code;
+	gchar *location;
+	gboolean completed;
+	GHashTable *messages_seen;
+	FuProgress *progress;
+} FuRedfishDevicePollCtx;
```

Comment:
I don't think we should export this; export a `fu_redfish_device_generic_poll_task_one()` that takes proper parameters.

Reply:
I'm not 100% sure of what you mean here:
- I use most of the context to push data
- moving that to pointer parameters would make a lot of parameters and be annoying
- the only other idea I have is to create a final class over `FuRedfishDevicePollCtx` (called something else) and interact with it using only setters from the outside. I find that a bit bulky, but I guess that would work

What were you thinking about exactly here?

Reply:
I think make it private again, and just copy and paste the `fu_redfish_device_poll_task` into `fu-redfish-hpe-device.c` as `fu_redfish_hpe_device_poll_task` -- you're duplicating ~15 lines of code and reducing complexity by a lot more than that.